File: CS50_Works/2-Problem_sets/c-Week2_Problem_Sets-c/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include "string.h"
#include "ctype.h"
#include<math.h>

//Functions Prototypes
float count_sentences(string plain_text, int Y);
float count_words(string plain_text, int Y);
float count_letters(string plain_text, int Y);

//Main Function

int main(void)
{
    string plain_text = get_string("Text : ");
    int Y = strlen(plain_text);

    float u = count_letters(plain_text, Y);
    float o = count_words(plain_text, Y);
    float p = count_sentences(plain_text, Y);
    float L = u * 100 / o;
    float S = p * 100 / o;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    printf("letters %f words %f sentence %f", count_letters(plain_text, Y), count_words(plain_text, Y), count_sentences(plain_text, Y));
    int Grade = round(index);
    if (Grade > 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    else if (Grade < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    else
    {
        printf("Grade %i\n", Grade);
        return 0;
    }


}

//Calculate Letters
float count_letters(string plain_text, int Y)
{
    int letters = 0;
    for (int i = 0 ; i < Y ; i++)
    {
        if (islower(plain_text[i]) != 0)
        {
            letters++;
        }
        else if (isupper(plain_text[i]) != 0)
        {
            letters++;
        }

    }
    return letters;
}

//Calculate Words
float count_words(string plain_text, int Y)
{
    int words = 1;
    char space = ' ';
    for (int i = 0 ; i < Y ; i++)
    {
        if (space == plain_text[i])
        {
            words++;
        }
    }
    return words;
}

//Calculate Sentences
float count_sentences(string plain_text, int Y)
{
    int sentence = 0;
    for (int i = 0 ; i < Y ; i++)
    {
        if (plain_text[i] == '!' || plain_text[i] == '?' || plain_text[i] == '.')
        {
            sentence++;
        }
    }
    return sentence;
}

File: CS50_Works/2-Problem_sets/c-Week6_Problem_Sets-c/sentimental-readability/readability.py
# TODO
from cs50 import get_string
from cs50 import get_int


def count_letters(Text):
    i = 0
    Letters = 0
    for i in range(0, len(Text), 1):
        if ord(f"{Text[i]}") in range(65, 91, 1):
            Letters = Letters + 1
        elif ord(f"{Text[i]}") in range(97, 123, 1):
            Letters = Letters + 1
    return Letters


def count_words(Text):
    Words = 0
    Words = len(Text.split())
    return Words


def count_sentences(Text):
    Sentences = 0
    for i in range(0, len(Text), 1):
        if Text[i] == '.' or Text[i] == '?' or Text[i] == '!':
            Sentences += 1
    return Sentences


def main():
    Text = get_string("Text : \n")

    Av_Letters = 100*count_letters(Text)/(count_words(Text))
    Av_Sentences = 100*count_sentences(Text)/(count_words(Text))

    index = 0.0588*Av_Letters - 0.296*Av_Sentences - 15.8

    print(index)

    if index > 16:
        print("Grade 16+\n")
    elif index < 1:
        print("Before Grade 1 \n")
    else:
        x = round(index)
        print(f"The Grade is {x}\n")

    print(count_letters(Text))
    print(count_words(Text))
    print(count_sentences(Text))


if __name__ == "__main__":
    main()

